lcls-twincat-motion/Library/POUs/Motion/PMPS/FB_MiscStatesErrorFFO.TcPOU

```diff
@@ -112,7 +112,7 @@ END_IF
 ffDebounce.i_xReset S= bFirstCycle;
 ffDebounce(
     i_xOK := nTripCount < 5,
-    i_xAutoReset := TRUE,
+    i_xAutoReset := nTripCount=0,
```

Comment:
I know this doesn't work, because `i_xAutoReset` is cached on first cycle.
https://github.com/pcdshub/lcls-twincat-pmps/blob/master/lcls-twincat-pmps/PMPS/MajorComponents/FastFaults/FB_FastFault.TcPOU

I was planning to try some combination of setting and un-setting the i_xReset input.

But maybe implementation-wise simplest is to just make the max count something like 10 so we don't have to change much.

Some side-thoughts:
- We don't need the manual reset-on-first-cycle if autoreset is true
- There are some unit tests specifically for this fault that might deserve a checkup

Reply:
You're right, as long as autoreset is on we have some edge case for bouncing the debouncer.

I think the main things to watch out for are:
- Causing conflicts with the existing logic that sets/resets `i_xReset` (we do need to set `i_xReset` on the first cycle, unfortunately, or we open in the faulting state always)
- Do we need to also set `i_xReset` back to false afterwards as past me did here? Was there a point to that?

Maybe trusting the old structure a bit and doing something like 
```
ffDebounce.i_xReset S= bFirstCycle OR nTripCount=0;
```
Would be simplest?

Let me know if I'm not making sense. I'm half expecting to be wrong here.

Reply:
Unfortunately the fast fault object starts in a faulted state and needs a reset. This usually isn't a problem because we autoreset everything.

Or, at least that's what I remember from implementing and testing this in the unit test suite a few years ago.